src/k8s/pkg/proxy/apiserver.go

```diff
@@ -68,18 +72,19 @@ func (p *APIServerProxy) watchForNewEndpoints(ctx context.Context, cancel func()
 		newEndpoints, err := getKubernetesEndpoints(ctx, p.KubeconfigFile)
 		switch {
 		case err != nil:
-			log.Println(fmt.Errorf("failed to retrieve kubernetes endpoints: %w", err))
+			log.Error(err, "Failed to retrieve Kubernetes endpoints")
 			continue
 		case len(newEndpoints) == 0:
-			log.Println("warning: empty list of endpoints, skipping update")
+			log.Info("Warning: empty list of endpoints, skipping update")
```

Comment:
should this be a warning level message instead of prefix this with "Warning"?

Reply:
Logs are either `.Info()` (with log levels) or `.Error()` (always printed)